BHoM_Engine/Compute/RecordEvent.cs

```diff
@@ -46,6 +46,31 @@ public static bool RecordEvent(string message, EventType type = EventType.Unknow
 
         /***************************************************/
 
+        [Description("Record an event with details of a C# exception within the BHoM logging system.")]
+        [Input("exception", "The C# exception being caught to provide the event and stack information for.")]
+        [Input("message", "An optional additional message which will be displayed first in the event log.")]
+        [Output("success", "True if the event has been successfully recorded as a BHoM Event.")]
+        public static bool RecordEvent(Exception exception, string message = "", EventType type = EventType.Unknown)
+        {
+            string exceptionMessage = "";
+
+            Exception e = exception;
+            while(e != null)
+            {
+                exceptionMessage += $"{e.Message}{Environment.NewLine}{Environment.NewLine}";
+                e = e.InnerException;
+            }
+
+            if (!string.IsNullOrEmpty(message))
+                message = $"{message}\n\n{exceptionMessage}";
+            else
+                message = exceptionMessage;
+
+            return RecordEvent(new Event { Message = message, StackTrace = exception.StackTrace, Type = EventType.Unknown });
```

Comment:
Do we need to pass in the `type` from the inputs here?
```suggestion

            return RecordEvent(new Event { Message = message, StackTrace = exception.StackTrace, Type = type  });
```